dsgrid/registry/dataset_registry_manager.py

```diff
@@ -191,16 +187,18 @@ def update(self, config, update_type, log_message, submitter=None):
             return self._update(config, submitter, update_type, log_message)
 
     def _update(self, config, submitter, update_type, log_message):
-        registry = self.get_registry_config(config.config_id)
-        old_key = ConfigKey(config.config_id, registry.version)
+        self._run_checks(config)
+        config_id = config.config_id
+        registry = self.get_registry_config(config_id)
+        old_key = ConfigKey(config_id, registry.version)
         version = self._update_config(config, submitter, update_type, log_message)
-        new_key = ConfigKey(config.config_id, version)
+        new_key = ConfigKey(config_id, version)
         self._datasets.pop(old_key, None)
         self._datasets[new_key] = config
 
         if not self.offline_mode:
-            self.sync_push(self.get_registry_directory(config.config_id))
-            self.sync_push(self.get_registry_data_directory(config.config_id))
+            self.sync_push(self.get_registry_directory(config_id))
```

Comment:
This confused me because in `def _register()`, `get_registry_directory()` uses `config.model.data_id`, then I realized config.config_id = config.model.data_id for dataset_manager.

Reply:
Yes, the parent class requires that `config_id` be a method so that it can have a consistent way of getting project_id/dataset_id/dimension_id/dimension_mapping_id. I'll make them consistent here.